modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/new-js/components/field-sets/FieldSetModal.js

```diff
@@ -0,0 +1,472 @@
+
+	const propagateFieldSet = usePropagateFieldSet();
+
+	const showPropagationModal = useCallback(
+		({onPropagate, updatedFieldSet}) => {
+			const modal = {
+				actionMessage: Liferay.Language.get('propagate'),
+				allowReferencedDataDefinitionDeletion: true,
+				fieldSetMessage: Liferay.Language.get(
+					'do-you-want-to-propagate-the-changes-to-other-objects-views-using-this-fieldset'
+				),
+				headerMessage: Liferay.Language.get('propagate-changes'),
+			};
+
+			const fieldNames = new Set(
+				updatedFieldSet.dataDefinitionFields.map(({name}) => name)
+			);
+
+			const hasRemovedAnyField = fieldSet.dataDefinitionFields.some(
+				({name}) => !fieldNames.has(name)
+			);
+
+			if (hasRemovedAnyField) {
+				modal.warningMessage = Liferay.Language.get(
+					'the-changes-include-the-deletion-of-fields-and-may-erase-the-data-collected-permanently'
+				);
+			}
+
+			return propagateFieldSet({fieldSet, modal, onPropagate});
+		},
+		[fieldSet, propagateFieldSet]
+	);
+
+	useEffect(() => {
+		if (fieldSet) {
+			const omit = (obj, props) => {
+				const result = {...obj};
+
+				props.forEach((prop) => {
+					delete result[prop];
+				});
+
+				return result;
+			};
+
+			const parseDataLayout = (dataLayout) => {
+				const {paginationMode, ...dataLayouts} = omit(dataLayout, [
+					'dataRules',
+					'dataLayoutPages',
+					'description',
+					'name',
+				]);
+
+				return {
+					dataLayout: dataLayouts,
+					paginationMode,
+				};
+			};
+
+			const parseDataDefinition = (dataDefinition) => {
+				const {
+					availableLanguageIds,
+					description,
+					name,
+					...dataDefinitions
+				} = omit(dataDefinition, [
+					'defaultLanguageId',
+					'defaultDataLayout',
+				]);
+
+				return {
+					availableLanguageIds,
+					dataDefinition: dataDefinitions,
+					description,
+					name,
+				};
+			};
+
+			const responseToJson = async (response) => {
+				if (response.ok) {
+					return await response.json();
+				}
+
+				throw response;
+			};
+
+			Promise.all([
+				fetch(
+					`${window.location.origin}/o/data-engine/v2.0/data-definitions/${fieldSet.id}`
+				).then(responseToJson),
+				fetch(
+					`${window.location.origin}/o/data-engine/v2.0/data-layouts/${fieldSet.defaultDataLayout.id}`
+				).then(responseToJson),
+			]).then(([dataDefinition, dataLayout]) => {
+				const ddmForm = getFieldSetDDMForm({
+					allowInvalidAvailableLocalesForProperty,
+					availableLanguageIds: fieldSet.availableLanguageIds,
+					editingLanguageId: fieldSet.defaultLanguageId,
+					fieldSet,
+					fieldTypes: config.fieldTypes,
+				});
+
+				new PagesVisitor(ddmForm.pages).mapFields((field) => {
+					field.label = field.label[fieldSet.defaultLanguageId];
+				});
+
+				setData({
+					...parseDataDefinition(dataDefinition),
+					...parseDataLayout(dataLayout),
+					dataDefinitionId: dataDefinition.id,
+					dataLayoutId: dataLayout.id,
+					name: dataDefinition.name,
+					pages: ddmForm.pages,
+				});
+			});
+		}
```

Comment:
You could replace this with `useData` which is used in the App to avoid some redundancies here, wasn´t it possible? what was the use case

Reply:
It wasn't working for us at first try so we decide move on and then try to use the useData, maybe we can check it again

Reply:
I see, what were the problems you have? in theory you would only have `dataDefinition`, `dataLayout` to use to complement the rest of the resource, see the verification that is also done in `App` because both can return `null` when the request is in progress.